api/src/main/java/jakarta/data/spi/provider/DataProvider.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) 2023 Contributors to the Eclipse Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+package jakarta.data.spi.provider;
+
+import java.lang.annotation.Annotation;
+import java.util.Set;
+
+import jakarta.data.exceptions.MappingException;
+
+/**
+ * <p>Jakarta Data providers can make themselves discoverable by
+ * Jakarta EE products via the Jakarta Data Provider SPI.
+ * Jakarta EE products with CDI support are thus able to automatically provide
+ * a CDI extension on behalf of the discovered Jakarta Data providers
+ * without those Jakarta Data providers themselves needing to depend on CDI.
```

Comment:
It did not understand this point, who use CDI extension and don't depend on CDI at the same time?

Reply:
IIUC, I think what @njr-11 meant is if a Jakarta EE runtime supports CDI, they can use CDI extension and they should be discovered by the Jakarta Data Provider. I think this might be a nice loosely coupled model. Jakarta Data itself does not depend on CDI but honours CDI. Maybe the sentence should be rephrased to make it clearer.

Reply:
Correct, it's the Jakarta EE product/runtime that supports CDI and can easily provide the CDI extension, and this gets us past the problem where some Jakarta Data providers don't want to depend on CDI themselves or directly interact with it at all.  I agree, I'll rephrase this to take out the confusing wording and try to state more directly, addressing both this review comment and the one before it and the one after at.  See commit https://github.com/jakartaee/data/pull/130/commits/3f27958aa294658ac1f9d96be110143da68b0558 which now states the purpose of this SPI more directly, and specifically for Jakarta EE profiles, and leaves the discussion about different options in Standalone and so forth to the spec doc where it is better written.